libraries/eosiolib/tester/eosio/tester.hpp

```diff
@@ -168,6 +168,55 @@ class test_chain {
                               const char* expected_except = nullptr);
    transaction_trace transact(std::vector<action>&& actions, const char* expected_except = nullptr);
 
+   /**
+    * Pushes a transaction with action onto the chain and converts its return value.
+    * If no block is currently pending, starts one.
+    */
+   template <typename Action, typename... Args>
+   auto action_with_return(const Action& action, Args&&... args) {
+      using Ret  = decltype(internal_use_do_not_use::get_return_type(Action::get_mem_ptr()));
+      auto trace = transact({action.to_action(std::forward<Args>(args)...)});
+      check(trace.action_traces[0].return_value.has_value(), "action did not return a value");
+      return check(convert_from_bin<Ret>(*trace.action_traces[0].return_value)).value();
+   }
+
+
+   template <typename Action, typename... Args>
+   auto act(const Action& action, Args&&... args) {
+      using Ret  = decltype(internal_use_do_not_use::get_return_type(Action::get_mem_ptr()));
+      auto trace = transact({action.to_action(std::forward<Args>(args)...)});
+      if constexpr ( !std::is_same_v<Ret,void> ) {
+         check(trace.action_traces[0].return_value.has_value(), "action did not return a value");
+         return check(convert_from_bin<Ret>(*trace.action_traces[0].return_value)).value();
+      } else {
+         return trace;
+      }
+   }
+
+   template <typename Action, typename... Args>
+   auto trace(const Action& action, Args&&... args) {
```

Comment:
I'm not sure I like the name trace.  It doesn't really make it clear what the function does nor what arguments it should take.